lib/beaker-template.rb

```diff
@@ -1,12 +1,25 @@
+require 'stringify-hash'
+require 'beaker-template/helpers'
+require 'beaker-template/version'
+
+
 module Beaker
   module DSL
-    module Helpers
-      module Template
-        require 'stringify-hash'
-        # include your modules from the beaker-template folder. Example below:
-        require 'beaker-template/helpers'
-        require 'beaker-template/version'
-      end
+    module Template
+
+      # include your modules from the beaker-template folder. Example below:
+      include Beaker::DSL::Template::Helpers
+
     end
   end
 end
+
+
+# Boilerplate DSL inclusion mechanism:
+# First we register our module with the Beaker DSL
+Beaker::DSL.register( Beaker::DSL::Template )
```

Comment:
@rick I used the `Beaker::DSL.register` pattern that you described in the old proposal in [beaker-pe](https://github.com/puppetlabs/beaker-pe/blob/master/lib/beaker-pe.rb). This code, which does work as an import mechanism to add the methods in the library, is commented out there, because currently we're using the beaker 2.y [import mechanism](https://github.com/puppetlabs/beaker/blob/master/lib/beaker/dsl.rb#L83), where we just pull it into beaker itself.

I'm looking to add this new way into the template, so that future libraries can use this same mechanism. I'm having trouble figuring out how to stub this method for spec testing, however.

One potential solution would be to not include the registration here, but to make a user include what they want in the DSL in their tests. I was hoping to avoid that, since it requires boilerplate code for the test writer. But maybe that's the customization part, where a user could include just what they want for a particular test rather than having all of these globally included?

Thought I'd push this up, & try to get some feedback on this. What do you think?